packages/commonwealth/client/scripts/state/api/comments/deleteComment.ts

```diff
@@ -75,18 +29,33 @@ const useDeleteCommentMutation = ({
 
   const { checkForSessionKeyRevalidationErrors } = useAuthModalStore();
 
-  return useMutation({
-    mutationFn: deleteComment,
+  return trpc.comment.deleteComment.useMutation({
     onSuccess: async (response) => {
+      // Important: we render comments in a tree, if the deleted comment is a
+      // leaf node, remove it, but if it has replies, then preserve it with
+      // [deleted] msg.
+      const canvasSignedData = await signDeleteComment(
```

Comment:
shouldn't this canvas-signed data be included in the payload (similar to the old hanlder)?

Reply:
canvas data is not being used in the command anywhere, that's why I decided to include it in the response instead... so caller of the mutation doesn't need to provide anything else other than the comment id... that's the main change I wanted you to review, let me know if that works